src/Analyzers/Analyzers/src/StartupAnalyzer.Diagnostics.cs

```diff
@@ -9,6 +9,7 @@ namespace Microsoft.AspNetCore.Analyzers
 {
     public partial class StartupAnalyzer : DiagnosticAnalyzer
     {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("MicrosoftCodeAnalysisReleaseTracking", "RS2008:Enable analyzer release tracking")]
```

Comment:
FYI @Pilchie. This seems onerous to do during ask mode hence suppressed. That said, it's not very clear what the benefit of doing the release tracking is. Would you be able to find out? https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md

Reply:
@mavasani - can you describe the customer benefit of enabling this analyzer?  The page linked above talks a lot about the what and the how, but very little about why we would want to do this as analyzer authors.

Reply:
Ah, sorry, I'll add this info to the doc. Analyzer release tracking is analogous to public API tracking, but for analyzer packages. It allows the analyzer authors to track which analyzers were part of each analyzer release, with their default severity + enabled by default state. For example, see https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/Core/AnalyzerReleases.Shipped.md and https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/Core/AnalyzerReleases.Unshipped.md. It basically tracks the how an analyzer package has evolved over time.

Additionally, this should eventually allow analyzers that ship as part of an SDK to enable warning waves support, wherein the end user can choose a specific analyzer release version in the project file, and the tooling will automatically disable all rules shipped for the first time after that release. We are using this mechanism to enable warning waves support for .NET analyzers that will ship with .NET5 SDK, though right now the tooling is very specific to this SDK.

Reply:
Filed https://github.com/dotnet/roslyn-analyzers/issues/3816